ReactAndroid/src/main/java/com/facebook/react/modules/toast/ToastModule.java

```diff
@@ -46,7 +56,30 @@ public String getName() {
   }
 
   @ReactMethod
-  public void show(String message, int duration) {
-    Toast.makeText(getReactApplicationContext(), message, duration).show();
+  public void show(String message, int duration, final int id) {
+    Toast toast = Toast.makeText(getReactApplicationContext(), message, duration);
+    toast.show();
+    mToasts.put(id, toast);
+    new Timer().schedule(new TimerTask() {
+      @Override
+      public void run() {
+        if (mToasts.containsKey(id)) {
+          mToasts.remove(id);
+        }
+      }
+    }, duration == Toast.LENGTH_SHORT ? SHORT_DELAY : LONG_DELAY);
```

Comment:
@astreet doesn't `Toast.getDuration` return `Toast.LENGTH_SHORT` or `Toast.LENGTH_LONG`? Which in turn will be 0 and 1?